app/controllers/users_controller.rb

```diff
@@ -70,6 +70,7 @@ def preferences
         render json: current_user.preferences
       end
     end
+    render layout: 'without_sidebar'
```

Comment:
This method has a section for what to do if HTML is requested, and a section for what to do if JSON is requested. The method is only allowed to have 1 render call. Putting a render call inside the HTML section will work fine, but putting it here outside both sections means that when JSON is requested (which already has its own render call) this will call render twice.

Moving this line inside the `format.html` block should remove the errors that are making the tests fail. It needs to be the last line in that block so that the variables being defined there are available to it.